furnace.c

```diff
@@ -147,12 +157,17 @@ command_handler(furnace_context_t* ctx, uint8_t* buffer, void (*feedback)(const
       const size_t msg_len = snprintf(msg, sizeof(msg), "pwm = %d\r\n", ctx->pwm_level);
       feedback(msg, msg_len);
   } else if (sscanf(buffer, "pwm %u", &arg) == 1) {
-    if (set_pwm_safe(ctx, arg) == 0){
+    int res;
+    if ((res = set_pwm_safe(FURNACE_FIRE_PIN, ctx, arg)) == 0){
       ctx->pilot.is_enabled = 0;
-    } else {
+    } else if ( res == 1){
       const char msg[] = "pwm argument too big!\r\n";
       const size_t msg_len = sizeof(msg)-1;
       feedback(msg, msg_len);
+    } else if ( res == -1){
+      const char msg[] = "set_pwm_safe: unexpected pin argument!\r\n";
+      const size_t msg_len = sizeof(msg)-1;
+      feedback(msg, msg_len);
```

Comment:
Consider adding a check to guard for all other, non-zero return values, just to be safe.